File: python/leap/leap.py
""" Given a year, report if it is a leap year. """

def is_even(year: int, div: int) -> bool:
    """ Validate year is evenly divisible """
    return year % div == 0

def leap_year(year: int) -> bool:
    """ Validate leap year """
    if is_even(year, 4) and is_even(year, 100):
        if is_even(year, 400):
            return True
        return False
    if is_even(year, 4):
        return True

    return False

File: lua/leap/leap.lua
---@param year integer
---@param number integer
---@return boolean
local divisible = function(year, number)
    return year % number == 0
end

---@param number integer
---@return boolean
local leap_year = function(number)
    if divisible(number, 400) then
        return true
    elseif divisible(number, 100) then
        return false
    elseif divisible(number, 4) then
        return true
    end
    return false
end

return leap_year
